src/main/java/org/isf/accounting/gui/PatientBillEdit.java

```diff
@@ -814,71 +922,50 @@ private JScrollPane getJScrollPaneTotal() {
-			jTableBigTotal.setMaximumSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
-			jTableBigTotal.setMinimumSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
-			jTableBigTotal.setPreferredSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
+			jTableBigTotal = new JTable(new JTableBigTotalModel());
+			sizeJTableBigTotal();
 		}
 		return jTableBigTotal;
 	}
 
+	private void sizeJTableBigTotal() {
+		jTableBigTotal.getColumnModel().getColumn(1).setMinWidth(CURRENCY_CODE_WIDTH);
+		jTableBigTotal.getColumnModel().getColumn(1).setMaxWidth(CURRENCY_CODE_WIDTH);
+		jTableBigTotal.getColumnModel().getColumn(2).setMinWidth(PRICE_WIDTH);
+		jTableBigTotal.getColumnModel().getColumn(2).setMaxWidth(PRICE_WIDTH);
+		jTableBigTotal.setMaximumSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
+		jTableBigTotal.setMinimumSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
+		jTableBigTotal.setPreferredSize(new Dimension(PANEL_WIDTH, BIG_TOTAL_HEIGHT));
+	}
+
+	private void setJTableBigTotal() {
+		jTableBigTotal.setModel(new JTableBigTotalModel());
+		sizeJTableBigTotal();
+	}
+
 	private JTable getJTableTotal() {
 		if (jTableTotal == null) {
-			jTableTotal = new JTable();
-			jTableTotal.setModel(new DefaultTableModel(new Object[][] {
-					{
-							"<html><b>"+MessageBundle.getMessage("angal.common.total.txt").toUpperCase()+"</b></html>",
-							currencyCod,
-							total}
-			},
-					new String[] {"","", ""}) {
-				private static final long serialVersionUID = 1L;
-				Class<?>[] types = new Class<?>[] { JLabel.class, JLabel.class, Double.class, };
-
-				@Override
-				public Class<?> getColumnClass(int columnIndex) {
-					return types[columnIndex];
-				}
-
-				@Override
-				public boolean isCellEditable(int row, int column) {
-					return false;
-				}
-			});
-			jTableTotal.getColumnModel().getColumn(1).setMinWidth(CURRENCY_CODE_WIDTH);
-			jTableTotal.getColumnModel().getColumn(1).setMaxWidth(CURRENCY_CODE_WIDTH);
-			jTableTotal.getColumnModel().getColumn(2).setMinWidth(PRICE_WIDTH);
-			jTableTotal.getColumnModel().getColumn(2).setMaxWidth(PRICE_WIDTH);
-			jTableTotal.setMaximumSize(new Dimension(PANEL_WIDTH, TOTAL_HEIGHT));
-			jTableTotal.setMinimumSize(new Dimension(PANEL_WIDTH, TOTAL_HEIGHT));
-			jTableTotal.setPreferredSize(new Dimension(PANEL_WIDTH, TOTAL_HEIGHT));
+			jTableTotal = new JTable(new JTableTotalModel());
+			sizeJTableTotal();
 		}
 		return jTableTotal;
 	}
 
+	private void setJTableTotal() {
+		jTableTotal.setModel(new JTableTotalModel());
+		sizeJTableTotal();
+	}
+
+	private void sizeJTableTotal() {
+		jTableTotal.getColumnModel().getColumn(1).setMinWidth(CURRENCY_CODE_WIDTH);
+		jTableTotal.getColumnModel().getColumn(1).setMaxWidth(CURRENCY_CODE_WIDTH);
+		jTableTotal.getColumnModel().getColumn(2).setMinWidth(PRICE_WIDTH);
+		jTableTotal.getColumnModel().getColumn(2).setMaxWidth(PRICE_WIDTH);
+		jTableTotal.setMaximumSize(new Dimension(PANEL_WIDTH, TOTAL_HEIGHT));
```

Comment:
wouldn't it be better to create the Dimension object once and refer to it 3 times?

Reply:
not sure what do you mean, can you make a code suggestion?

Reply:
No; you aren't looking at the complete file contents:

![image](https://user-images.githubusercontent.com/1105445/223496873-769683a2-aa84-4cb3-8dd7-bf41857ee22d.png)

![image](https://user-images.githubusercontent.com/1105445/223496933-dfafb13e-49fb-4fb4-937b-6bb35d157f38.png)

![image](https://user-images.githubusercontent.com/1105445/223496994-b43b488b-e7eb-46ca-9794-f9d72c834a8b.png)

Reply:
Alright, done! Thanks!